docs/openlaw-object/README.md

```diff
@@ -0,0 +1,2141 @@
+
+Returns an array of validation errors as strings. An empty array `[]` will be returned if there are no errors.
+
+### hasMissingInputs
+
+Check if a contract has any missing variable inputs after it has been validated by the [`validateContract` method](#validatecontract).
+
+```scala
+hasMissingInputs(
+  result: ValidationResult
+): Boolean
+```
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `result` | [`ValidationResult`](#validationresult) | **Required.** The object returned from the [`validateContract` method](#validatecontract). |
+
+Example
+
+```js
+// see examples above for #execute and #executeForReview for parameters
+const executionResult = Openlaw.execute(compiledTemplate.compiledTemplate, {}, params);
+const validationResult = Openlaw.validateContract(executionResult.executionResult);
+Openlaw.hasMissingInputs(validationResult);
+```
+
+**Response**
+
+Returns `true` if validated contract has any missing variable inputs.
+
+### getMissingInputs
+
+List any missing variable inputs of a contract after it has been validated by the [`validateContract` method](#validatecontract).
+
+```scala
+getMissingInputs(
+  result: ValidationResult
+): js.Array[String]
+```
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `result` | [`ValidationResult`](#validationresult) | **Required.** The object returned from the [`validateContract` method](#validatecontract). |
+
+Example
+
+```js
+// see examples above for #execute and #executeForReview for parameters
+const executionResult = Openlaw.execute(compiledTemplate.compiledTemplate, {}, params);
+const validationResult = Openlaw.validateContract(executionResult.executionResult);
+Openlaw.getMissingInputs(validationResult);
+```
+
+**Response**
+
+Returns an array of missing variable inputs as strings. An empty array `[]` will be returned if there are no missing variable inputs.
+
+Example
+
+```js
+[
+  "Company Name",
+  "Advisor Email"
+]
+```
+
+### getTemplateName
+
+Get name of template.
+
+```scala
+getTemplateName(
+  templateDefinition: TemplateDefinition
+): String
+```
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `templateDefinition` | `TemplateDefinition` | **Required.** An object nested within a [`TemplateExecutionResult` object](#templateexecutionresult). |
+
+Example
+
+```js
+// see examples above for #execute and #executeForReview for parameters
+const executionResult = Openlaw.execute(compiledTemplate.compiledTemplate, {}, params);
+const templateDefinition = executionResult.executionResult.templateDefinition;
+Openlaw.getTemplateName(templateDefinition);
+```
+
+**Response**
+
+Returns name of template as string.
+
+Example
+
+```js
+"Advisor Agreement"
+```
+
+### getAgreements
+
+Get all agreements linked to a template for rendering. A [deal](/markup-language/#deals) template will have one or more agreements. A non-deal template will have only one agreement.
```

Comment:
Is there a situation where a deal would have just one agreement or is it always more than one? If not, then -> A deal will have more than one agreement

Reply:
It's possible for a deal template to be linked to just one agreement. For example, the Employee Onboarding deal could end just having the Employee Offer Letter because the other agreements are conditional. Also, technically you can set up a deal to just include one agreement (even though the usefulness of a deal is bundling multiple agreements).

Reply:
Ah, I see. Good points!